packages/react/src/components/tag.tsx

```diff
@@ -38,28 +53,43 @@ export interface TagProps extends HTMLAttributes<HTMLElement> {
 }
 
 export function Tag({
-  size = 'medium',
+  size = 'md',
   variant = 'readonly',
+  icon = null,
   className,
+  severity,
   children,
   asChild = false,
   ...rest
 }: TagProps) {
+  const [isActive, setIsActive] = useState<boolean>(true);
+  const hasIcon = icon !== null
   const classes = clsx(
-    `r-tag r-tag--${size}`,
+    `r-tag r-tag--${size} r-tag--${severity}`,
     {
       /**
        * The readonly variant is the default and doesn't need a modifier class.
        */
       [`r-tag--${variant}`]: variant !== 'readonly',
+      'r-tag--has-icon': hasIcon,
+      [`r-tag--${severity}--inactive`]: variant === 'selectable' && !isActive,
     },
-    className
+    className,
   )
   const Comp = asChild ? Slot : 'span'
+
+  const handleClick = () => {
+    setIsActive((prev) => !prev);
+  }
+ 
   return (
-    <div className='text-red-500'>
+    <div onClick={handleClick}>
       <Comp className={classes} {...rest}>
+      {hasIcon && <span className='r-tag--icon-container'>{icon}</span>}
         {children}
+        {variant === 'dismissible' && (
+          <XIcon className='r-tag--x' />
```

Comment:
Don't use a modifier class in this situation. Think also the name can be changed to something more telling. See my suggestion as an example.

```suggestion
          <XIcon className='r-tag__remove-icon' />
```